src/tool/xmeta/cpp/models/base_model.h

```diff
@@ -26,12 +31,65 @@ namespace xlang::xmeta
     struct method_model;
     struct event_model;
 
-    using class_type_semantics = std::variant<
+    struct formal_parameter_model;
+    struct symbol_table;
+    struct compilation_unit;
+    struct type_ref;
+
+    enum class compilation_error
+    {
+        passed = 0,
+        symbol_exists,
+        accessor_exists,
+        method_cannot_be_overloaded
+    };
+
+
+    enum class method_association
+    {
+        None,
+        Property,
+        Event,
+        Constructor
+    };
+
+    enum class simple_type
```

Comment:
These are refered to as "fundamental" types in the type system spec: https://github.com/microsoft/xlang/blob/master/design_notes/XDN03%20-%20xlang%20Type%20System.md#fundamental-types